### PR TITLE
Issue #11: Shuttle Bus navigation

### PR DESCRIPTION
**Pull request by makchamp**
**Issue #11: Shuttle Bus navigation**

- Added a new Fragment and ViewModel for shuttle.
- The fragment can be accessed from the side nav menu
- The shuttle view model uses the navigation repository to get the shuttle times
- Fixed an issue with the shuttle times not being retrieved at all from the database.
- Separating the Shuttle Fragment into 2 Tabs (SGW TO LOY, and LOY TO SGW) using ViewPage2
- The data for the departure times is shown in a table
- Generating directions between campuses

### CODE REVIEW

File: app/src/main/java/com/droidhats/campuscompass/views/ShuttleFragment.kt [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        }
+    }
+
+    private fun observeSGWShuttle() {
+        viewModel.getSGWShuttleTime().observe(viewLifecycleOwner, Observer { sgwShuttleTimes ->
+            val table = root.findViewById<TableLayout>(R.id.shuttleTimesTable)
+            for (dataRow in sgwShuttleTimes) {
+                val tableRow = TableRow(root.context)
+                val dayText = TextView(root.context)
+                /*  var layoutParams = TableRow.LayoutParams(
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/views/ShuttleFragment.kt]: we should try not to comment out code

File: app/src/main/java/com/droidhats/campuscompass/MainActivity.kt [hunk 2]
```diff
@@ -15,11 +15,12 @@ import androidx.navigation.fragment.NavHostFragment
 import androidx.navigation.ui.setupWithNavController
 import com.droidhats.campuscompass.views.CalendarFragment
 import com.droidhats.campuscompass.views.MapFragment
+import com.droidhats.campuscompass.views.ShuttleFragment
 import com.droidhats.campuscompass.views.SplashFragment
 import com.google.android.material.navigation.NavigationView
 
 
-class MainActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener{
+class MainActivity : AppCompatActivity(){
```
Comment (NicholasNagy) [inline app/src/main/java/com/droidhats/campuscompass/MainActivity.kt]: we should leave a space between the bracket and curly bracket

Comment (makchamp) [reply app/src/main/java/com/droidhats/campuscompass/MainActivity.kt]: this has been removed

File: app/src/main/java/com/droidhats/campuscompass/views/ShuttleFragment.kt (hunk 1 above)
Comment (makchamp) [reply app/src/main/java/com/droidhats/campuscompass/views/ShuttleFragment.kt]: all good

**Comment by NicholasNagy** [comment]
I saw the changes on the shuttle fragment, everything looks good to merge now.

**Pull request merged**